### DISCUSSION TITLE
How to correctly form query with dateFiled option

### DISCUSSION THREAD

**Discussion by amit-lf** [freelawproject/courtlistener#4197] [Q&A] [answered]
The query 
"tenant landlord dispute"  dateFiled:[2024-01-01 TO 2024-04-30]

which has a corresponding API call - https://www.courtlistener.com/?q=%22tenant%20landlord%20dispute%22%20%20dateFiled%3A%5B2024-01-01%20TO%202024-04-30%5D&type=o&order_by=dateFiled%20desc&stat_Precedential=on

does not restrict results between the specified dateFiled.  

How can I correctly formulate the query to achieve this please?

**Comment by mlissner** [comment, marked as answer]
I think this is a bit out of sync because of our transition from Solr to Elastic search, sorry. 

A simple solution is to use the `filed_after` and `filed_before` parameters. Something like: 

https://www.courtlistener.com/?q=%22tenant%20landlord%20dispute%22&type=o&order_by=score%20desc&stat_Precedential=on&filed_after=07%2F09%2F2024&filed_before=07%2F10%2F2024

If you need to do a different field and want to use the query box, then I think the distinction is that one (Solr, I think) uses ISO-8601 with the time, and the other (elastic) uses ISO-8601 without the time.

**Reply by mlissner** [reply]
Sorry, one other thing. If you're setting up a new system, you should probably try to use v4 of the API. It's not yet documented and therefore not fully launched, but it's code complete and the docs are coming soon. There are very few changes between the two, but it has more features and is more performant.

**Reply by amit-lf** [reply]
Thanks Michael. The filed_before and filed_after restricts the records in the date range. 

Yes setting up a new system, will reach out to you at the right time. Will switch to v4 as performance is important.

**Reply by mlissner** [reply]
Great! We do like to set up API agreements with folks using the APIs for commercial purposes, so let's discuss that when you have a moment! My email is mike@free.law if you want to get started that way.
